src/GenSymIO.chpl

```diff
@@ -1421,48 +1451,51 @@ module GenSymIO {
-                       * correspond to chars shuffled to the previous locale (idx-1) and 
-                       * (2) generate a new, corresponding Strings segments list. 
-                       */  
-                      var charList : list(uint(8));
-                      var segmentsList : list(int);
-                      var valuesList : list(uint(8));
-                     
-                      (charList, segmentsList) = sliceToValuesAndSegments(A.localSlice(locDom)); 
-                       
-                      /*
-                       * Check to see if previous locale (idx-1) ends with a complete string.
-                       * If not, then the leading slice of this string was used to complete
-                       * the last string in the previous locale, so slice those chars.
+                       * Check to see if previous locale (idx-1) ends with a null character.
+                       * If not, then the left shuffle slice of this locale was used to complete
+                       * the last string in the previous locale, so slice those chars from 
+                       * this locale and create a new, corresponding charList.
                        */
                       if !endsWithCompleteString(idx-1) {
-                          (valuesList, segmentsList) = 
-                                             adjustForLeadingSlice(leadingSliceIndex, charList);
+                          var localStart = locDom.first;
+                          var localLeadingSliceIndex = localStart + shuffleLeftIndex;
+                          var leadingCharArray = adjustCharArrayForLeadingSlice(localLeadingSliceIndex, 
+                                         A.localSlice(locDom),locDom.last,idx);
+                          charList = new list(leadingCharArray);  
+                          gsLogger.info(getModuleName(),getRoutineName(),getLineNumber(),
+                                  'adjusted locale %i for left shuffle to locale %i'.format(
+                                         idx,idx-1));
                       } else {
-                          valuesList = charList;
-                      }
+                          charList = new list(A.localSlice(locDom));
+                          gsLogger.info(getModuleName(),getRoutineName(),getLineNumber(),
+                                  'no left shuffle from locale %i to locale %i'.format(
+                                         idx,idx-1));
+                      } 
+                      
+                      // Generate the segments list now that the char list is finalized
+                      var segmentsList = generateFinalSegmentsList(charList,idx);
 
                       // Write the finalized valuesList and segmentsList to the hdf5 group
-                      writeStringsToHdf(myFileID, group, valuesList, segmentsList);
+                      writeStringsToHdf(myFileID, idx, group, charList, segmentsList);
```

Comment:
I noticed these lines are now common to both if/else clauses (which is great!):
```
// Generate the segments list now that the char list is finalized
var segmentsList = generateFinalSegmentsList(charList,idx);

// Write the finalized valuesList and segmentsList to the hdf5 group
writeStringsToHdf(myFileID, idx, group, charList, segmentsList);
```
If you want to do further clean-up they could be moved outside the if/else and consolidated.

Reply:
Hmm... it's also happening in the parent if block starting at line 1238 (executed around line 1385).  In theory if you moved the declartion of `charList` outside of that block you could then move the `generateFinalSegmentsList` and `writeStringsToHdf` outside of those parent if/else blocks, but it might be a bit dicey and beyond the scope of this PR.  Sorry just thinking out-loud here as I work through this section of the code ;)

Reply:
Agreed, good call, will make that change